diff --git a/node_modules/solid-refresh/babel.js b/node_modules/solid-refresh/babel.js
index 9358054..f293a14 100644
--- a/node_modules/solid-refresh/babel.js
+++ b/node_modules/solid-refresh/babel.js
@@ -71,7 +71,7 @@ module.exports = ({ types: t }) => {
           ];
         }
 
-        path.replaceWithMultiple(replacement);
+        path.replaceWithMultiple(replacement).forEach(declaration => path.scope.registerDeclaration(declaration));
       }
     }
   };
